torchinductor/lowering.py

```diff
@@ -2522,6 +2540,36 @@ def copy_(dst, src, non_blocking=False):
     return mutate_to(dst, src)
 
 
+@make_pointwise
+def floordiv(a, b):
+    return ops.floordiv(a, b)
+
+
+@make_pointwise
+def truncdiv(a, b):
+    return ops.truncdiv(a, b)
+
+
+@register_lowering(aten.div.Tensor_mode)
+def div_mode(a, b, rounding_mode=None):
+    both_integer = is_integer_type(a) and is_integer_type(b)
```

Comment:
this doesn't correctly handle situations when one of the inputs is integer and one is boolean, and trunc or floor mode is set (pytorch produces integer result in this case)

Reply:
Should be solved by including `is_boolean_dtype` checking in `is_integer_type`.

Reply:
If both inputs are boolean, pytorch errors out

Reply:
Is there a spec somewhere for this kind of typing rules? Or is it basically trying and getting the results? For this specific both boolean problem, I wonder if we can relax the implementation in inductor? After all, the front end should error our before reaching here, and there are other ops currently implemented without very rigid checking.

Reply:
If I write the following test,
```
    def test_div6(self):
        def fn(a, b):
            return (
                aten.div(a, b, rounding_mode=None),
                aten.div(a, b, rounding_mode="floor"),
                aten.div(a, b, rounding_mode="trunc"),
            )

        # treat boolean as integer
        self.common(
            fn, (torch.ones([8, 8], dtype=torch.bool), torch.ones([8, 8], dtype=torch.bool))
        )
```
it does error out before even reaching my changed code with,
```
E       RuntimeError: "div_floor_cpu" not implemented for 'Bool'

../pytorch-release/torch/_ops.py:175: RuntimeError
```

@ngimel , what do you think?

Reply:
Yeah it errors out with eager error. However, imagine in future we would be tracing with fake tensors instead of real ones, then this error wouldn't happen and it would be on inductor to produce correct behavior, so it still makes sense to have necessary checks in inductor, even if we can't test them.